File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-lesson-model.js


import ViewLesson from "../views/view-lesson";
import ResponseCommentModel from "./response-comment-model";

class ResponseLessonModel extends ViewLesson {
    constructor() {
        super();
        this.AverageStar = null;
        this.MyStars = null;
        this.TotalStar = null;
        this.NumberComments = null;
        this.TopComments = null;
        this.IsMarked = null;
        this.Categories = [];
    }

    init() {
        return new ResponseLessonModel();
    }

    copy(entity) {
        super.copy(entity);
        if (this.TopComments?.length > 0){
            this.TopComments = this.TopComments.map(function(com){
                let comm = new ResponseCommentModel();
                comm.copy(com);
                return comm;
            });
        }
        return this;
    }
}


export default ResponseLessonModel;

File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseLessonModel.cs
using KnowledgeSharingApi.Domains.Annotations.Converters.ResponseModel;
using KnowledgeSharingApi.Domains.Interfaces.ModelInterfaces.ApiResponseModelInterfaces;
using KnowledgeSharingApi.Domains.Models.Entities.Tables;
using KnowledgeSharingApi.Domains.Models.Entities.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    //[JsonConverter(typeof(ResponseUserItemConverter))]
    public class ResponseLessonModel : ViewLesson, IResponseLessonModel
    {
        public double? AverageStars { get; set; }
        public double? MyStars { get; set; }
        public int TotalStars { get; set; }

        public int NumberComments { get; set; }
        public List<ResponseCommentModel> TopComments { get; set; } = [];
        public bool IsMarked { get; set; }
        public List<Category> Categories { get; set; } = [];
    }
}
